### ISSUE TITLE
[BUG] Plotting randomly stuck without any error message.

### ISSUE THREAD

**Issue by AkiraDX** [Chia-Network/chia-blockchain#2986]
Hi...

I noticed a new issue recently. The plotting will randomly stuck at some point
without any progress and any error message.

I see the windows process is still there but just not moving at all.

"Starting phase 3/4: Compression from tmp files into .....
Compressing tables 1 and 2
	Bucket 0 uniform sort. Ram: 7.865GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 1 uniform sort. Ram: 7.865GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 2 uniform sort. Ram: 7.865GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 3 uniform sort. Ram: 7.865GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 4 uniform sort. Ram: 7.865GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 5 uniform sort. Ram: 7.865GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 6 uniform sort. Ram: 7.865GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 7 uniform sort. Ram: 7.865GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 8 uniform sort. Ram: 7.865GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 9 uniform sort. Ram: 7.865GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 10 uniform sort. Ram: 7.865GiB, u_sort min: 1.250GiB, qs min: 0.313GiB"

- OS: Windows 10
- OS Version/Flavor: Windows 10 21H1
- CPU: AMD Ryzen 9 5900X

**Comment by MasterAssailant** [comment]
Hello, I am the author of issue #2951. I believe I am having a similar issue as you do. Mind if I ask what is the brand and model of the temp plot drive (i.e. an SSD) are you using?

**Comment by AkiraDX** [comment]
> 
> 
> Hello, I am the author of issue #2951. I believe I am having a similar issue as you do. Mind if I ask what is the brand and model of the temp plot drive (i.e. an SSD) are you using?

Gigabyte NVMe4 500GB SSD x 2 - RAID0

**Comment by MasterAssailant** [comment]
What about the plotting configurations u using? (i.e. buffer size and thread number)? 
Are you plotting in parallel or in a queue?
And how often do you face this issue? For me, I come across this issue once around 2 to 3 plots...

**Comment by a250** [comment]
Have exactly the same issue.
i have ran single plotting twice. Previously it stuck on 6%. Then was my second attempt,  after restart and deleting all temporary files, and now i stuck on 11% for several hours (3+ hours).

there is no parameters like buffer size and thread numbers in GUI when you run single plot. 

i'm under Ubuntu 20.04.02
SSD ~500GB in total and about 450GB was free at the started time

I checked the performance parameters and saw, that i'm in swap

MEM 2.0/3.7GB
SWP 1.0/2.0GB

**Comment by C4** [comment]
I have have run into similar issues (on two different plotting machines) in **both** cases it turned out to be RAM that had gone bad. I had done every diagnostic thinking it was my plotting drive (nvme in my case) and was about to give up until I read an obscure post about checking the RAM. 

It may be worth a try to run a mem test on your ram (https://www.memtest86.com/)

**Comment by AkiraDX** [comment]
> 
> 
> What about the plotting configurations u using? (i.e. buffer size and thread number)?
> Are you plotting in parallel or in a queue?
> And how often do you face this issue? For me, I come across this issue once around 2 to 3 plots...

I'm running in Multiple parallel queues.

my config is 16 threads and other config is default.

Happen randomly like once a day or 2 days once.
It's completely random.

**Comment by razamataz** [comment]
> I have have run into similar issues (on two different plotting machines) in **both** cases it turned out to be RAM that had gone bad. I had done every diagnostic thinking it was my plotting drive (nvme in my case) and was about to give up until I read an obscure post about checking the RAM.
> 
> It may be worth a try to run a mem test on your ram (https://www.memtest86.com/)
 I have have run into similar issues
Ram test is OK
any other suggestions?

**Comment by MasterAssailant** [comment]
Another question, are you guys using the machine just for plotting, or are u using it say for work and are running other programs at the same time?

**Comment by MasterAssailant** [comment]
> I have have run into similar issues
> Ram test is OK
> any other suggestions?

May I ask what SSD brand and model are u using for storing temp plot files?

**Comment by razamataz** [comment]
Only for Chia.Windows is new.
Samsung 970 EVO Plus

**Comment by MasterAssailant** [comment]
To those who are still facing this issue, another brilliant friend found out a possible solution in solving this issue (https://github.com/Chia-Network/chia-blockchain/issues/2951#issuecomment-835732970). Feel free to try this out!

**Comment by AkiraDX** [comment]
This issue is getting worse and worse with new v1.1.15.
It stuck many times in a day. The issue was not this bad
with v1.1.3 and earlier.

I can't use the workaround by end task the stucked chia.exe process
then reclaim the temporary space then start a new process anymore
since v1.1.4.

If I end task stucked chia.exe then it will end task the all chia.exe that's
still plotting fine. This kind of behavior was not happened in v1.1.3
and earlier.
